examples/common/chip-app-server/Server.cpp

```diff
@@ -45,26 +45,21 @@ using namespace ::chip::DeviceLayer;
 
 namespace {
 
-class ServerCallback : public SecureSessionMgrDelegate
+class ServerCallback : public ExchangeContextDelegate
 {
 public:
-    void OnMessageReceived(const PacketHeader & header, const PayloadHeader & payloadHeader, Transport::PeerConnectionState * state,
-                           System::PacketBuffer * buffer, SecureSessionMgrBase * mgr) override
+    void OnMessageReceived(ExchangeContext * ec, const PacketHeader & packetHeader, uint32_t protocolId, uint8_t msgType, System::PacketBuffer * buffer) override
```

Comment:
server (AKA device) will receive the allocated `ExchangeContext` here. The context is allocated by `ExchangeManager` when receiving a message with `Initiator` flags set.

Reply:
Yes, this is all fine so far.

But the low-level send functions need to deal with transactions initiated by the server too.